### PR TITLE
Enable `rose-suite.conf[template variables]`

### PR DESCRIPTION
**Pull request by wxtim**
This PR plus [Cylc-flow PR 4073](https://github.com/cylc/cylc-flow/pull/4073) address #25.

Added functional test to ensure that this works properly.
Modified the logic ensuring the _only one_ template variables section is allowed.

Tests will fail online until [Cylc-flow PR 4073](https://github.com/cylc/cylc-flow/pull/4073) is merged.

### CODE REVIEW

File: cylc/rose/utilities.py [hunk 1]
```diff
@@ -57,19 +57,16 @@ def get_rose_vars_from_config_node(config, config_node, environ):
...
         )
-    elif 'jinja2:suite.rc' in config_node.value:
-        templating = 'jinja2'
-    elif 'empy:suite.rc' in config_node.value:
-        templating = 'empy'
-    if templating:
+    elif len(defined_sections) == 1:
+        templating, = list(defined_sections)
+    if templating is not None:
         config['templating_detected'] = templating
```
Comment (oliver-sanders) [inline cylc/rose/utilities.py:70]: I think this messes with the reported value of `templating_detected`?

Possible fix:

```suggestion
    elif len(defined_sections) == 1:
        templating, = defined_sections
        if templating != 'template variables':
            config['templating_detected'] = templating.replace(':suite.rc', '')
```

Comment (wxtim) [reply cylc/rose/utilities.py:70]: It does, I just changed the value of `templating_detected` all the way through and into cylc-flow. I think it makes the code easier to follow.

**Review by oliver-sanders — COMMENTED** [review]
Looks good, tested as working.

```console
$ cat flow.cylc 
#!Jinja2

[scheduling]
    [[graph]]
        R1 = a

# @( ROSE_SUITE_VARIABLES )
# {{ ROSE_SUITE_VARIABLES }}
$ # test jinja2 processing
$ cylc view --process . --stdout --debug
...
2021-02-16T16:21:03Z DEBUG - Processing with Jinja2
2021-02-16T16:21:03Z DEBUG - Setting Jinja2 template variables:
	+ CYLC_VERSION=8.0a3.dev
	+ FLOAT=4.2
	+ INT=42
	+ LIST=[1, True, 'b']
	+ ROSE_ORIG_HOST=deep-thought
	+ ROSE_SITE=
	+ ROSE_SUITE_VARIABLES={'ROSE_ORIG_HOST': 'deep-thought',
	'ROSE_VERSION': '2.0a1', 'ROSE_SITE': '', 'LIST': [1, True, 'b'], 'FLOAT': 4.2,
	'STR': 'string', 'INT': 42, 'ROSE_SUITE_VARIABLES': {...}}
	+ ROSE_VERSION=2.0a1
	+ STR=string
[scheduling]
    [[graph]]
        R1 = a
# @( ROSE_SUITE_VARIABLES )
# {'ROSE_ORIG_HOST': 'deep-thought', 'ROSE_VERSION': '2.0a1', 'ROSE_SITE': '', 'LIST': [1, True, 'b'], 'FLOAT': 4.2, 'STR': 'string', 'INT': 42, 'ROSE_SUITE_VARIABLES': {...}}
$ # test empy processing
$ sed -i 's/Jinja2/empy/' flow.cylc 
$ cylc view --process . --stdout --debug
...
2021-02-16T16:21:13Z DEBUG - Processing with EmPy
[scheduling]
    [[graph]]
        R1 = a
# {'ROSE_ORIG_HOST': 'deep-thought', 'ROSE_VERSION': '2.0a1', 'ROSE_SITE': '', 'LIST': [1, True, 'b'], 'FLOAT': 4.2, 'STR': 'string', 'INT': 42, 'ROSE_SUITE_VARIABLES': {...}}
# {{ ROSE_SUITE_VARIABLES }}
```

I did spot a(n) (unrelated) difference in the logging between Jinja2 and empy where Jinja2 lists the template variables and empy doesn't?

**Review by oliver-sanders — APPROVED** [review]

**Comment by oliver-sanders** [comment]
Could do with a PR or issue on Rose to update the `*:suite.rc` things to `template variables` (ensuring the relevant shebang is set in the right places).

**Review by datamel — APPROVED** [review]
I have read the code and run the tests. LGTM, thanks Tim!

**Comment by wxtim** [comment]
> Could do with a PR or issue on Rose to update the *:suite.rc things to template variables (ensuring the relevant shebang is set in the right places).

@oliver-sanders I'm not sure I understand what you are asking for here - Which files does Rose check for shebangs?

**Comment by oliver-sanders** [comment]
Rose2019 automatically adds shebangs.

Rose2 will/does also do this but only if the `suite.rc` file is present (for back compat) once upgraded to `flow.cylc` users will have to do this themselves. Actually, have we done this or is there an issue 😨.

So we will need to make sure that any examples in the Rose docs explicitly state the shebangs they need at the top of the `flow.cylc` files.

**Pull request merged**